OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -210,5 +220,25 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr, World world)
 		}
 
 		public string GetCursor(World world, CPos xy, MouseInput mi) { return "default"; }
+
+		IEnumerable<Order> TryClearBlockers(World world, CPos topLeft)
+		{
+			var allTiles = FootprintUtils.Tiles(world.Map.Rules, building, buildingInfo, topLeft).ToArray();
+			var neightborTiles = Util.ExpandFootprint(allTiles, true).Except(allTiles)
+				.Where(world.Map.Contains).ToList();
+
+			var blockers = allTiles.SelectMany(world.ActorMap.GetUnitsAt)
+				.Where(a => a.Owner == producer.Owner && a.IsIdle && a.HasTrait<Mobile>());
+
+			foreach (var blocker in blockers)
+			{
+				var availableCells = neightborTiles.Where(t => blocker.Trait<Mobile>().CanEnterCell(t));
```

Comment:
Can you check if this is empty, and `continue` if so?
Not doing so is causing auditory feedback that is actually invalid.

>Moving out!
When there is in fact nowhere to go.